### DISCUSSION TITLE
Hss payload Yocto

### DISCUSSION THREAD

**Discussion by Shivaraj199** [polarfire-soc/polarfire-soc-documentation#200] [Linux build systems for PolarFire SoC] [answered]
Hello,
Is it possible i can include   hss-payload-generator-native recipe with the Yocto Build,
I know in documents the hss binaries are shipped with Yocto image, is t possible to use the lib  of hss payload inside the image ?
is yes how can i access it?

**Comment by gsriker77** [comment]
Yes, hss-payload-generator-native is independent source for which recipe can be added in Yocto.
At present we are providing only binaries in image. One can write recipe by adding below dependency libs and generate  hss-payload-generator executable.
	-lyaml \r 
	-lelf \
	-lz \
	-lcrypto \
If you are interested, We can create recipe and provide you.
Thanks

**Reply by Shivaraj199** [reply]
Hello @gsriker77 ,
Thank you for the solution,
yes please it would be of great help if you could provide with the recipe, there as some libraries that cannot be cross compiled as eg libtorch as it precompiled binaries for x86 , so we need to compile it from the source on the target. 
That said it would be great if we can use the libraries of hss payload such as drivers for our application. If the recipe could provide with the source of hss-payload-generator-native instead of binaries  would be very helpful as we can use the lib for our custom application and compile it on the target.

**Comment by gsriker77** [comment, marked as answer]
https://github.com/polarfire-soc/meta-polarfire-soc-yocto-bsp/blob/master/recipes-bsp/hss-payload-generator-native/hss-payload-generator-native.bb

**Comment by Shivaraj199** [comment]
Hello Thank you but this  recipe  i tried to already include in image and it throws error 

`ERROR: Task do_populate_sdk in /home/yocto-dev/meta-polarfire-soc-yocto-bsp/recipes-core/images/mpfs-dev-cli.bb rdepends upon non-existent task do_package_write_ipk in /home//home/yocto-dev/meta-polarfire-soc-yocto-bsp/recipes-bsp/hss-payload-generator-native/hss-payload-generator-native.bb
ERROR: Command execution failed: 1`

**Reply by hughbreslin** [reply]
Hey @Shivaraj199 did you get this working? From reading over this is the goal to have the HSS payload generator included in the root FS of the Linux image?

**Reply by Shivaraj199** [reply]
Hey @hughbreslin 

No It didnt work, I was just looking into, if the HSS payload is been included the roootfs as there was a recipe that was given the BSP I wanted to look into the payload and play a bit around it, but when i tried to bake the recipe alone it was not compiling.
